### PR TITLE
WINDUPRULE-513 org.apache.camel.spring.Main has been moved

### PR DESCRIPTION
**Pull request by johnpoth**
https://issues.redhat.com/browse/WINDUPRULE-513

To run tests: `mvn -DrunTestsMatching=xml-moved-components clean test`

Thanks !

### CODE REVIEW

File: rules-reviewed/camel3/camel2/xml-moved-components.windup.xml [hunk 1]
```diff
@@ -237,6 +237,47 @@
...
+                <xmlfile as="dependencies-block" in="pom.xml"
+                         matches="//*[text() = 'org.apache.camel.spring.Main' and /m:project/m:dependencies[count(m:dependency/m:artifactId[text()='camel-spring-main'])=0]]">
+                    <namespace prefix="m" uri="http://maven.apache.org/POM/4.0.0"/>
+                </xmlfile>
+            </when>
+            <perform>
+                <iteration>
+                    <hint title="`org.apache.camel.spring.Main` class has been moved" effort="1" category-id="mandatory">
+                        <message>The class `org.apache.camel.spring.Main` has been moved out of `org.apache.camel:camel-spring` and into its own JAR named `org.apache.camel:camel-spring`.
+                            `org.apache.camel:camel-spring` has to be added as a dependency to your project pom.xml file
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/xml-moved-components.windup.xml:251]: > has been moved out of `org.apache.camel:camel-spring` and into its own JAR named `org.apache.camel:camel-spring`.
Shouldn't it be "into its own JAR named `org.apache.camel:camel-spring-main` ?

Comment (jonathanvila) [reply rules-reviewed/camel3/camel2/xml-moved-components.windup.xml:251]: tbh, I'm not sure what you are searching for in this rule . Searching a node in POM.XML with this text `org.apache.camel.spring.Main` ?

File: rules-reviewed/camel3/camel2/xml-moved-components.windup.xml [hunk 2]
```diff
@@ -237,6 +237,47 @@
...
+                    <namespace prefix="m" uri="http://maven.apache.org/POM/4.0.0"/>
+                </xmlfile>
+                <javaclass references="org.apache.camel.spring.Main">
+                    <location>IMPORT</location>
+                </javaclass>
+            </when>
+            <perform>
+                <iteration over="dependencies-block">
+                    <hint title="`org.apache.camel.spring.Main` class has been moved" effort="1" category-id="mandatory">
+                        <message>The class `org.apache.camel.spring.Main` has been moved out of `org.apache.camel:camel-spring` and into its own JAR named `org.apache.camel:camel-spring`.
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/xml-moved-components.windup.xml:272]: > has been moved out of org.apache.camel:camel-spring and into its own JAR named org.apache.camel:camel-spring.

Shouldn't it be "into its own JAR named org.apache.camel:camel-spring-main ?

File: rules-reviewed/camel3/camel2/xml-moved-components.windup.xml [hunk 3]
```diff
@@ -237,6 +237,47 @@
...
+                            `org.apache.camel:camel-spring` has to be added as a dependency to your project pom.xml file
+                        </message>
+                        <link href="https://camel.apache.org/manual/latest/camel-3x-upgrade-guide.html#_main_in_camel_spring"
+                              title="Camel 3 - Main camel-spring"/>
+                    </hint>
+                </iteration>
+            </perform>
+        </rule>
+        <rule id="xml-moved-components-00015">
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/xml-moved-components.windup.xml:259]: Will this rule need to be refactored to groovy as the others that were having an `xmlfile`+`javaclass` condition ?

**Comment by johnpoth** [comment]
Hi @jonathanvila thanks for reviewing! Actually this belong in 3.2 migration guide ... closing

**Pull request closed without merging**